src/core/public/application/application_service.test.ts

```diff
@@ -905,6 +910,30 @@ describe('#start()', () => {
       expect(setupDeps.redirectTo).toHaveBeenCalledWith('/test/app/baseApp');
     });
   });
+
+  describe('navigateToUrl', () => {
+    it('calls `redirectTo` when the url is not parseable', async () => {
+      parseAppUrlMock.mockReturnValue(undefined);
```

Comment:
Why not use a real call? The logic there is rather simple.

Reply:
We could remove the mock and perform a real call but as `parseAppUrl` is heavily tested in its own suite, I think current approach is slightly better in term of testing isolation.